lib/derivative_rodeo/generators/plain_text_generator.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module DerivativeRodeo
+  module Generators
+    ##
+    # Generate the word coordinates (as JSON) from the given input_uris.
+    #
+    # @note Assumes that we're receiving a HOCR file (generated via {HocrGenerator}).
+    class PlainTextGenerator < BaseGenerator
+      self.output_extension = "plain_text.txt"
+
+      ##
+      # @param output_location [StorageLocations::BaseLocation]
+      # @param input_tmp_file_path [String] the location of the file that we can use for processing.
+      #
+      # @return [StorageLocations::BaseLocation]
+      #
+      # @see #requisite_files
+      def build_step(output_location:, input_tmp_file_path:, **)
+        output_location.with_new_tmp_path do |output_tmp_file_path|
+          convert_to_coordinates(path_to_hocr: input_tmp_file_path, path_to_plain_text: output_tmp_file_path)
+        end
+      end
+
+      private
+
+      ##
+      # @param path_to_hocr [String]
+      # @param path_to_plain_text [String]
+      # @param service [#call, Services::ExtractWordCoordinatesFromHocrSgmlService]
+      def convert_to_coordinates(path_to_hocr:, path_to_plain_text:, service: Services::ExtractWordCoordinatesFromHocrSgmlService)
```

Comment:
Wondering if this service could be a class_attribute?  That allows for configuration and has the same dependency injection behavior.

Reply:
looks like i would need
```rb
require_relative '../services/extract_word_coordinates_from_hocr_sgml_service'

...
      self.output_extension = "plain_text.txt"

        class_attribute :service, default: Services::ExtractWordCoordinatesFromHocrSgmlService
...
      def convert_to_coordinates(path_to_hocr:, path_to_plain_text:)
        hocr_html = File.read(path_to_hocr)
        File.open(path_to_plain_text, "w+") do |file|
          file.puts self.class.service.call(hocr_html).to_text
        end
      end
...
```

Is that what you had in mind?